src/state-mutation.js

```diff
@@ -18,8 +18,13 @@ const updateReverseRelationship = (
         const relCase = [1, 2]
           .map(i => pluralize(entity.get('type'), i))
           .find(r => foreignEntity.hasIn(['relationships', r]));
+        const relCount = foreignEntity.getIn(['relationships', relCase, 'data']);
+        const isset = relCount &&
+          relCount.getIn &&
+          relCount.getIn([0]) &&
+          relCount.getIn([0]).count();
```

Comment:
Hi @matiaszanolli
Im gonna have a look at this (Testing it durring this week) and will get back to you for sure. I can agree with you, that all these checks is not sexy at all, but if it would solve this case for now, untill we find a more stable solution we might just go with this, since its a very needed feature of the API.

I thank you for your time spend already on this :) 👍